pandas/core/indexes/base.py

```diff
@@ -5375,6 +5362,7 @@ def identical(self, other) -> bool:
                 for c in self._comparables
             )
             and type(self) == type(other)
+            and self.dtype == other.dtype
```

Comment:
Previously the index type could differentiate e.g. `Index([1, 2, 3], dtype=object)` and `Index([1, 2, 3], dtype="int64")` (the latter actually being a `NumericIndex`), but we now need to be more precise.